File: Java/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }

        StringBuilder result = new StringBuilder();

        for (int j = 0; j < strs[0].length(); j++) {
            char c = strs[0].charAt(j);
            for (int i = 1; i < strs.length; i++) {
                if (j >= strs[i].length() || strs[i].charAt(j) != c) {
                    return result.toString();
                }
            }
            result.append(c);
        }

        return result.toString();
    }
}

File: C++/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        string ret;

        for (int j = 0; j < strs[0].length(); ++j) {
            char c = strs[0][j];
            for (int i = 1; i < strs.size(); ++i) {
                if (j >= strs[i].length() || strs[i][j] != c) {
                    return ret;
                }
            }
            ret += c;
        }
        return ret;
    }
};
